polyply/tests/test_apply_links.py

```diff
@@ -327,6 +327,35 @@ def test_match_link_and_residue_atoms_fail(example_meta_molecule,
                          [[(1, 4, 1)], [(6, 9, 1)], [(1, 4, 5, 6, 1)]],
                          [[(0, 0)], [(1, 0)], [(2, 0)]]
                         ),
+                        # 8 - simple check single residue but both nodes are the same
+                        ([[(0, {'name': 'BB'}),
+                           (1, {'name': 'BB1'}),
+                           (2, {'name': 'SC1'})]],
+                         [{0: 0, 1: 0, 2:0}],
+                         ['angles'],
+                         [[vermouth.molecule.Interaction(atoms=(0, 1, 2),
+                                                        parameters=['1', '124', '500'],
+                                                        meta={})]],
+                         [[]],
+                         [[]],
+                         [[(0, 1, 2, 1)]],
+                         [[(0, 0)]]
+                        ),
+                        # 9 - test node get scheduled for removal
+                        ([[(0, {'name': 'BB'}),
+                           (1, {'name': 'BB1'}),
+                           (2, {'name': 'SC1'}),
+                           (3, {'name': 'SC2', 'replace': {'atomname': None}})]],
+                         [{0: 0, 1: 0, 2:0, 3:0}],
+                         ['angles'],
+                         [[vermouth.molecule.Interaction(atoms=(0, 1, 2),
+                                                        parameters=['1', '124', '500'],
+                                                        meta={})]],
+                         [[]],
+                         [[]],
+                         [[(0, 1, 2, 1)]],
+                         [[(0, 0)]]
```

Comment:
Add a testcase where an interaction has to get removed.

Reply:
I don't think polyply removes interactions at the moment in the sense of what !bonds I think would do

Reply:
No. Assume a bond between atoms 1 and 3, and atom 1 (or 3) gets removed: what happens to the bond?

Reply:
the bond get's remove but not here; If a link were to have a bond where an atom would be scheduled for removal then it would be added to the bond-list. Not sure how that actually works in martinize2. I would say a link having a bond with a node that get's removed is undefined ?!

Reply:
Hmmn, ok.

In vermouth the behaviour is defined: interactions get removed by `Molecule.remove_node`.

Reply:
ahh ok it's the same for polyply but they all get removed in the end. The case is actually covered by the test_gen_params test case.